doc/source/guide/filters.rst

```diff
@@ -0,0 +1,143 @@
+``.gcov`` files created by ``gcov``.
+This is useful mostly when running gcov yourself,
+and then invoking gcovr with :option:`-g/--use-gcov-files<gcovr --use-gcov-files>`.
+But these filters also apply when gcov is launched by gcovr.
+
+
+Speeding up coverage data search
+--------------------------------
+
+The :option:`--exclude-directories<gcovr --exclude-directories>` filter is used
+while searching for raw coverage data (or for existing ``.gcov`` files when
+:option:`-g/--use-gcov-files<gcovr --use-gcov-files>` is active).
+This filter is matched against directory paths, not file paths.
+If a directory matches,
+all its contents (files and subdirectories) will be excluded from the search.
+For example, consider this build directory::
+
+    build/
+    ├─ main.o
+    ├─ main.gcda
+    ├─ main.gcno
+    ├─ a/
+    │  ├─ awesome_code.o
+    │  ├─ awesome_code.gcda
+    │  └─ awesome_code.gcno
+    └─ b/
+       ├─ better_code.o
+       ├─ better_code.gcda
+       └─ better_code.gcno
+
+If we run ``gcovr --exclude-directories 'build/a$'``,
+this will exclude anything in the ``build/a`` directory
+but will use the coverage data for ``better_code.o`` and ``main.o``.
+
+This can speed up gcovr when you have a complicated build directory structure.
+Consider also using the :option:`search_paths <gcovr search_paths>`
+or :option:`--object-directory<gcovr --object-directory>` arguments to specify
+where gcovr starts searching.
+If you are unsure which directories are being searched,
+run gcovr in :option:`-v/--verbose<gcovr --verbose>` mode.
+
+For each found coverage data file gcovr will invoke the ``gcov`` tool.
+This is typically the slowest part,
+and other filters can only be applied *after* this step.
+In some cases, parallel execution with the :option:`-j<gcovr -j>` option
+might be helpful to speed up processing.
+
+
+Filters for symlinks
+--------------------
+
+Gcovr matches filters against *real paths*
+that have all their symlinks resolved.
+E.g. consider this project layout::
+
+    /home/you/
+    ├─ project/  (pwd)
+    │  ├─ src/
+    │  ├─ relevant-library/ -> ../external-library/
+    │  └─ ignore-this/
+    └─ external-library/
+       └─ src/
+
+.. compare the filter-relative-lib test case
+
+Here, the ``relevant-library``
+has the real path ``/home/you/external-library``.
+
+To write a filter that includes both ``src/`` and ``relevant-library/src/``,
+we cannot use ``--filter relevant-library/src/``
+because that contains a symlink.
+Instead, we have to use an absolute path to the real name::
+
+    gcovr --filter src/ --filter /home/you/external-library/src/
+
+or a relative path to the real path::
+
+    gcovr --filter src/ --filter '\.\./external-library/src/'
+
+.. note::
+    This section discusses symlinks on Unix systems.
+    The behavior under Windows is unclear.
+    If you have more insight,
+    please update this section by submitting a pull request
+    (see our :ref:`contributing guide <contributing>`).
```

Comment:
I think this can be removed because they behave the same way. One note should be added, a windows symbolic link isn't a symlink. A symlink is the same think like a junction.